services/src/modules/directives/merge-with-stub.ts

```diff
@@ -0,0 +1,23 @@
+import { SchemaDirectiveVisitor } from 'graphql-tools';
+import { GraphQLField, defaultFieldResolver } from 'graphql';
+import { gql } from 'apollo-server-core';
+import { inject } from '../arguments-injection';
+import { RequestContext } from '../context';
+
+export class MergeWithStubDirective extends SchemaDirectiveVisitor {
+  visitFieldDefinition(field: GraphQLField<unknown, RequestContext>) {
+    const { value } = this.args;
+    const originalResolve = field.resolve || defaultFieldResolver;
+
+    field.resolve = async (parent, args, context, info) => {
+      const originalValue = await originalResolve.call(field, parent, args, context, info);
+      const stubValue = typeof value === 'string' ? inject(value, parent, args, context, info) : value;
+
+      return { ...originalValue, ...stubValue };
```

Comment:
Maybe we can add optional parameter to the directive named `deepMerge` of type boolean with default false and if it's true use lodash or ramda merge...

Reply:
Sounds reasonable, but it won't be a breaking change so we can wait until we need it (since I'm not 100% sure there will be a use case for it)